src/app/network/download.ts

```diff
@@ -166,55 +166,103 @@ async function getRequiredFileMetadataWithAuth(
 }
 
 export function downloadFile(params: IDownloadParams): Promise<ReadableStream<Uint8Array>> {
+  let connectionLost = false;
+  const timeoutMs = 10000;
+
+  const connectionLostListener = () => {
+    connectionLost = true;
+    window.removeEventListener('offline', connectionLostListener);
+  };
+
+  window.addEventListener('offline', connectionLostListener);
+  const timeoutId = setTimeout(() => {
+    if (!navigator.onLine) {
+      connectionLost = true;
+    }
+  }, timeoutMs);
```

Comment:
Better not to add more complexity to the download function, having the `downloadManager` you can add the detection there and this function will only take care of downloading the file